### PR TITLE
fuzzing: Add config validation fuzzer

### PR DESCRIPTION
**Pull request by AdamKorcz**
This PR adds a fuzzer that creates a config and validates it.

Once this fuzzer is merged, I will make the necessary changes on the OSS-fuzz side to run this fuzzer continuously.

[x] Configuration Infrastructure
[ ] Docs
[ ] Installation
[ ] Networking
[ ] Performance and Scalability
[ ] Policies and Telemetry
[x] Security
[ ] Test and Release
[ ] User Experience
[ ] Developer Infrastructure

Pull Request Attributes

Please check any characteristics that apply to this pull request. 

[x] Does not have any [user-facing](https://github.com/istio/istio/tree/master/releasenotes#when-to-add-release-notes) changes. This may include CLI changes, API changes, behavior changes, performance improvements, etc.

### CODE REVIEW

File: tests/fuzz/config_validation_fuzzer.go [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// nolint: golint
+package fuzz
+
+import (
+	fuzz "github.com/AdaLogics/go-fuzz-headers"
```
Comment (howardjohn) [inline tests/fuzz/config_validation_fuzzer.go:19]: I would have expected a go.mod change resulting from this?

Comment (AdamKorcz) [reply tests/fuzz/config_validation_fuzzer.go:19]: Hereby added.

**Review by howardjohn — APPROVED** [review]

**Comment by howardjohn** [comment]
/retest

known test flake

**Pull request merged**